### ISSUE TITLE
Issues Using DVCLive with PyTorch Lightning Project

### ISSUE THREAD

**Issue by Hamptonjc** [iterative/dvclive#475]
Though I have used DVC a bit for data tracking, I am new to using it for experiment tracking. Following the tutorials, I am using the DVCLiveLogger with my PyTorch Lightning project. I have set save_dvc_exp = True. The problems I am facing:

- After a training run completes, it takes ~1 hour to finally exit the training script. In my repo, I am storing several different (dvc tracked) datasets, each around 2G. I briefly get a message about DVC "building data objects" for the datasets. Is this approach okay (i.e. storing several different datasets)? Is there anyway I can speed this process up?

- The training metrics I am logging in PyTorch Lightning seem to be getting incorrectly grouped. For example, in the produced metrics.json, my 'val_model_error' metric is parsed into:
```json
"val": {
        "error": {
            "model": 0.1
        }
}
```

Any suggestions?

**Comment by daavoo** [comment]
> After a training run completes, it takes ~1 hour to finally exit the training script. In my repo, I am storing several different (dvc tracked) datasets, each around 2G. I briefly get a message about DVC "building data objects" for the datasets. Is this approach okay (i.e. storing several different datasets)? Is there anyway I can speed this process up?

The approach is okay.

DVCLive is calling `dvc exp save` under the hood. That operation involves a `dvc commit` and I suspect that is the place it's taking so long. 
We are most likely doing something unnecessary there.

Could you share the output of:

```
pip install viztracer
dvc exp save --viztracer --viztracer-depth 15
```

You don't need to wait until full completion, interrupting the process should still generate a profile.

> The training metrics I am logging in PyTorch Lightning seem to be getting incorrectly grouped. For example, in the produced metrics.json, my 'val_model_error' metric is parsed into:

How are you logging the metric? 

By default, PyTorch Lightning adds prefixes and suffixes to the logged metric like (`train_{metric_name}_epoch`) and we try to standardize the names of metrics and use a nested structure because it is better for our UIs (VSCode, Studio):

https://github.com/iterative/dvclive/blob/3308c6806e0e33d59f1960f2a4a465b5293f8ecc/src/dvclive/utils.py#L89-L95

It appears that you are encountering the logic in a case where it is not necessary. 
We could probably add a flag to opt-out of this standardization and preserve the original metric names.

**Comment by dberenbaum** [comment]
> By default, PyTorch Lightning adds prefixes and suffixes to the logged metric like (`train_{metric_name}_epoch`) and we try to standardize the names of metrics and use a nested structure because it is better for our UIs (VSCode, Studio):

@daavoo Maybe the logic is too generic in the case of lightning? In other frameworks, we replace specific prefixes like `val_` but don't assume any `_` should be split. I think lightning maybe needed something more complex than that, but it seems like splitting every `_` may be too unexpected.

**Comment by Hamptonjc** [comment]
![image](https://user-images.githubusercontent.com/41594631/221944481-07b4b560-9a83-4566-a89b-96b43a2ee3ca.png)

![image](https://user-images.githubusercontent.com/41594631/221945480-037dae12-e744-436b-9d4f-a7761ffda9dd.png)

@daavoo Indeed, it seems dvc commit is the issue.

I am logging inside of my lightning module like so:

```python
def validation_step(self, ...):
    ...
    self.log('val_model_error', model_error.item(),
        on_epoch=True, prog_bar=False, logger=True)
```

**Comment by daavoo** [comment]
> @daavoo Indeed, it seems dvc commit is the issue.

I have sent a patch (#9093) which should remove some unnecessary overhead in commit.

The rest of the overhead (in the first screenshot) I suspect is the same underlying issue as https://github.com/iterative/dvc/issues/9085 and we will be working on it.

> I am logging inside of my lightning module like so:

As per @dberenbaum comment, we need to revisit the current standardization logic for lightning but I think it is still useful to have an option to disable it. Will sent a patch tomorrow for the disabling

**Comment by dberenbaum** [comment]
> @daavoo Maybe the logic is too generic in the case of lightning? In other frameworks, we replace specific prefixes like `val_` but don't assume any `_` should be split. I think lightning maybe needed something more complex than that, but it seems like splitting every `_` may be too unexpected.

I opened a PR in #480. I'm not sure it's worth merging because:

1. It will break the existing grouping for use cases like the one above, which might surprise users.
2. Lightning leaves so much of the naming up to the user that only the `_step` and `_epoch` suffixes are actually enforced. Prefixes like `train_`, `val_`, and `test_` are consistenly suggested but not enforced anywhere.

However, it probably doesn't hurt much to try to standardize when those conventions are followed.

**Comment by daavoo** [comment]
> Prefixes like train_, val_, and test_ are consistenly suggested but not enforced anywhere.

😞  All this time I thought it was inserted by the framework based on whether it is inside `training_step/validation_step` method or something

**Comment by daavoo** [comment]
Closing for now. https://github.com/iterative/dvc/pull/9093 and https://github.com/iterative/dvclive/pull/480 where merged

**Comment by Hamptonjc** [comment]
@daavoo The tool is usable for my project now, thanks!!
